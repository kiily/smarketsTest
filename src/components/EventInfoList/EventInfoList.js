import React from 'react';
import EventInfo from './EventInfo/EventInfo';
import { format } from 'date-fns';

import './EventInfoList.scss';

const eventInfoList = (props) => {
  return (
    <div className="EventInfoList-list-container">
      <ul className="event-info-list">
        {props.events.map((event, index) => {
          const eventType = event.type.split('_')[0];
          const parentCompetition = event.parent.name;
          const teams = event.name.split(' vs. ');
          const startTime = new Date(event.start_datetime);
          console.log("TCL: eventInfoList -> startTime", startTime)
          return (
            <EventInfo isTall={index === 0} key={event.id} sport={eventType} competition={parentCompetition}
              teams={teams} startTime={format(startTime, 'DD/MM/YYYY hh:mm')}/>
          );
        })}
      </ul>
    </div>
  );
};

export default eventInfoList;
