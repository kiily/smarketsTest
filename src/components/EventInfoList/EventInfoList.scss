@import '../../styles/_variables.scss';

.EventInfoList-list-container {
    .event-info-list {
        display: grid;
        grid-gap: 10px;
        grid-auto-flow: row dense;
        grid-template-columns: repeat(auto-fill, minmax(18rem, 1fr));
        grid-auto-rows: minmax(131px, auto);
    }
}