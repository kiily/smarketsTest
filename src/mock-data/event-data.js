export const events = [
  {
    'actual_start_datetime': null,
    'bettable': true,
    'created': '2019-04-16T10:09:18.291988Z',
    'description': null,
    'display_order': 0,
    'end_date': null,
    'full_slug': '/sport/football/championship/2019/04/30/millwall-fc-vs-bristol-city-fc',
    'hidden': false,
    'parent': {
      'name': 'Championship'
    },
    'id': '1925896',
    'inplay_enabled': true,
    'modified': '2019-04-16T10:09:18.291988Z',
    'name': 'Millwall vs. Bristol City',
    'parent_id': '939285',
    'short_name': 'MWL vs. BRC',
    'slug': 'millwall-fc-bristol-city-fc-2019-04-30',
    'special_rules': null,
    'start_date': '2019-04-30',
    'start_datetime': '2019-04-30T18:45:00Z',
    'state': 'upcoming',
    'type': 'football_match'
  },
  {
    'actual_start_datetime': null,
    'bettable': true,
    'created': '2019-04-17T10:10:36.025237Z',
    'description': null,
    'display_order': 0,
    'end_date': null,
    'full_slug': '/sport/football/championship/2019/05/01/swansea-vs-derby',
    'hidden': false,
    'parent': {
      'name': 'Championship'
    },
    'id': '1934371',
    'inplay_enabled': true,
    'modified': '2019-04-17T10:10:36.025237Z',
    'name': 'Swansea City vs. Derby County',
    'parent_id': '939285',
    'short_name': 'SWA vs. DER',
    'slug': 'swansea-derby-2019-05-01',
    'special_rules': null,
    'start_date': '2019-05-01',
    'start_datetime': '2019-05-01T18:45:00Z',
    'state': 'upcoming',
    'type': 'football_match'
  },
  {
    'actual_start_datetime': null,
    'bettable': true,
    'created': '2019-04-18T10:11:56.786364Z',
    'description': null,
    'display_order': 0,
    'end_date': null,
    'full_slug': '/sport/football/europe-uefa-champions-league/2019/04/30/tottenham-vs-afc-ajax',
    'hidden': false,
    'parent': {
      'name': 'Champions League'
    },
    'id': '1943310',
    'inplay_enabled': true,
    'modified': '2019-04-18T10:11:56.786364Z',
    'name': 'Tottenham Hotspur vs. Ajax',
    'parent_id': '929817',
    'short_name': 'TOT vs. AJA',
    'slug': 'tottenham-afc-ajax-2019-04-30',
    'special_rules': null,
    'start_date': '2019-04-30',
    'start_datetime': '2019-04-30T19:00:00Z',
    'state': 'upcoming',
    'type': 'football_match'
  },
  {
    'actual_start_datetime': null,
    'bettable': true,
    'created': '2019-04-18T10:12:03.638683Z',
    'description': null,
    'display_order': 0,
    'end_date': null,
    'full_slug': '/sport/football/europe-uefa-champions-league/2019/05/01/fc-barcelona-vs-liverpool-fc',
    'hidden': false,
    'parent': {
      'name': 'Champions League'
    },
    'id': '1943311',
    'inplay_enabled': true,
    'modified': '2019-04-18T10:12:03.638683Z',
    'name': 'Barcelona vs. Liverpool',
    'parent_id': '929817',
    'short_name': 'BAR vs. LIV',
    'slug': 'fc-barcelona-liverpool-fc-2019-05-01',
    'special_rules': null,
    'start_date': '2019-05-01',
    'start_datetime': '2019-05-01T19:00:00Z',
    'state': 'upcoming',
    'type': 'football_match'
  },
  {
    'actual_start_datetime': null,
    'bettable': true,
    'created': '2019-04-20T10:06:14.610452Z',
    'description': null,
    'display_order': 0,
    'end_date': null,
    'full_slug': '/sport/football/premier-league/2019/05/04/wolverhampton-vs-fulham-fc',
    'hidden': false,
    'parent': {
      'name': 'Premier League'
    },
    'id': '1958062',
    'inplay_enabled': true,
    'modified': '2019-04-20T10:06:14.610452Z',
    'name': 'Wolverhampton Wanderers vs. Fulham',
    'parent_id': '943086',
    'short_name': 'WOL vs. FUL',
    'slug': 'wolverhampton-fulham-fc-2019-05-04',
    'special_rules': null,
    'start_date': '2019-05-04',
    'start_datetime': '2019-05-04T14:00:00Z',
    'state': 'upcoming',
    'type': 'football_match'
  }
];
