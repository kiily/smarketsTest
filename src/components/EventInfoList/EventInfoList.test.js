import React from 'react';
import { create } from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

import EventInfoList from './EventInfoList';
import { events } from '../../mock-data/event-data';

describe('EventInfoList component', () => {
  let component;
  beforeEach(() => {
    component = create(<BrowserRouter><EventInfoList events={events} /></BrowserRouter>);
  });

  it('should match the snapshot when rendered with props', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('should render a list item tag (li) per event (5) events', () => {
    const rootInstance = component.root;
    const list = rootInstance.findByType('ul');
    expect(list.children).toHaveLength(5);
  });
});
