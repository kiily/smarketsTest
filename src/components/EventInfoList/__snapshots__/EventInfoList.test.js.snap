// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventInfoList component should match the snapshot when rendered with props 1`] = `
<ul
  className="EventInfoList-list-container"
>
  <li
    className="EventInfo-tile tall"
  >
    <a
      href="/event-detail/1925896"
      onClick={[Function]}
    >
      <div
        className="link-overlay"
      />
    </a>
    <div
      className="info-container"
    >
      <div
        className="category-container"
      >
        <div
          className="category-text-container"
        >
          <span
            className="category-text"
          >
            football
          </span>
          <span
            className="separator"
          >
            &gt;
          </span>
          <span
            className="category-text"
          >
            Championship
          </span>
        </div>
        <div
          className="event-state"
        >
          <span
            className="state-text"
          >
            upcoming
          </span>
        </div>
      </div>
      <div
        className="info-title"
      >
        <div
          className="teams"
        >
          <div
            className="team"
          >
            <span
              className="team-name"
            >
              Millwall
            </span>
          </div>
          <div
            className="team"
          >
            <span
              className="team-name"
            >
              Bristol City
            </span>
          </div>
        </div>
      </div>
      <div
        className="event-date"
      >
        Start date: 
        30/04/2019 19:45
      </div>
    </div>
  </li>
  <li
    className="EventInfo-tile "
  >
    <a
      href="/event-detail/1934371"
      onClick={[Function]}
    >
      <div
        className="link-overlay"
      />
    </a>
    <div
      className="info-container"
    >
      <div
        className="category-container"
      >
        <div
          className="category-text-container"
        >
          <span
            className="category-text"
          >
            football
          </span>
          <span
            className="separator"
          >
            &gt;
          </span>
          <span
            className="category-text"
          >
            Championship
          </span>
        </div>
        <div
          className="event-state"
        >
          <span
            className="state-text"
          >
            upcoming
          </span>
        </div>
      </div>
      <div
        className="info-title"
      >
        <div
          className="teams"
        >
          <div
            className="team"
          >
            <span
              className="team-name"
            >
              Swansea City
            </span>
          </div>
          <div
            className="team"
          >
            <span
              className="team-name"
            >
              Derby County
            </span>
          </div>
        </div>
      </div>
      <div
        className="event-date"
      >
        Start date: 
        01/05/2019 19:45
      </div>
    </div>
  </li>
  <li
    className="EventInfo-tile "
  >
    <a
      href="/event-detail/1943310"
      onClick={[Function]}
    >
      <div
        className="link-overlay"
      />
    </a>
    <div
      className="info-container"
    >
      <div
        className="category-container"
      >
        <div
          className="category-text-container"
        >
          <span
            className="category-text"
          >
            football
          </span>
          <span
            className="separator"
          >
            &gt;
          </span>
          <span
            className="category-text"
          >
            Champions League
          </span>
        </div>
        <div
          className="event-state"
        >
          <span
            className="state-text"
          >
            upcoming
          </span>
        </div>
      </div>
      <div
        className="info-title"
      >
        <div
          className="teams"
        >
          <div
            className="team"
          >
            <span
              className="team-name"
            >
              Tottenham Hotspur
            </span>
          </div>
          <div
            className="team"
          >
            <span
              className="team-name"
            >
              Ajax
            </span>
          </div>
        </div>
      </div>
      <div
        className="event-date"
      >
        Start date: 
        30/04/2019 20:00
      </div>
    </div>
  </li>
  <li
    className="EventInfo-tile "
  >
    <a
      href="/event-detail/1943311"
      onClick={[Function]}
    >
      <div
        className="link-overlay"
      />
    </a>
    <div
      className="info-container"
    >
      <div
        className="category-container"
      >
        <div
          className="category-text-container"
        >
          <span
            className="category-text"
          >
            football
          </span>
          <span
            className="separator"
          >
            &gt;
          </span>
          <span
            className="category-text"
          >
            Champions League
          </span>
        </div>
        <div
          className="event-state"
        >
          <span
            className="state-text"
          >
            upcoming
          </span>
        </div>
      </div>
      <div
        className="info-title"
      >
        <div
          className="teams"
        >
          <div
            className="team"
          >
            <span
              className="team-name"
            >
              Barcelona
            </span>
          </div>
          <div
            className="team"
          >
            <span
              className="team-name"
            >
              Liverpool
            </span>
          </div>
        </div>
      </div>
      <div
        className="event-date"
      >
        Start date: 
        01/05/2019 20:00
      </div>
    </div>
  </li>
  <li
    className="EventInfo-tile "
  >
    <a
      href="/event-detail/1958062"
      onClick={[Function]}
    >
      <div
        className="link-overlay"
      />
    </a>
    <div
      className="info-container"
    >
      <div
        className="category-container"
      >
        <div
          className="category-text-container"
        >
          <span
            className="category-text"
          >
            football
          </span>
          <span
            className="separator"
          >
            &gt;
          </span>
          <span
            className="category-text"
          >
            Premier League
          </span>
        </div>
        <div
          className="event-state"
        >
          <span
            className="state-text"
          >
            upcoming
          </span>
        </div>
      </div>
      <div
        className="info-title"
      >
        <div
          className="teams"
        >
          <div
            className="team"
          >
            <span
              className="team-name"
            >
              Wolverhampton Wanderers
            </span>
          </div>
          <div
            className="team"
          >
            <span
              className="team-name"
            >
              Fulham
            </span>
          </div>
        </div>
      </div>
      <div
        className="event-date"
      >
        Start date: 
        04/05/2019 15:00
      </div>
    </div>
  </li>
</ul>
`;
